src/PDPVerifier.sol

```diff
@@ -597,25 +598,27 @@ contract PDPVerifier is Initializable, UUPSUpgradeable, OwnableUpgradeable {
         }
     }
 
-    function calculateProofFee(uint256 setId, uint256 estimatedGasFee) public view returns (uint256) {
+    function calculateProofFee(uint256 setId) public view returns (uint256) {
         uint256 rawSize = 32 * challengeRange[setId];
-        (uint64 filUsdPrice, int32 filUsdPriceExpo) = getFILUSDPrice();
+        return calculateProofFeeForSize(rawSize);
+    }
 
-        return PDPFees.proofFeeWithGasFeeBound(
-            estimatedGasFee, filUsdPrice, filUsdPriceExpo, rawSize, block.number - dataSetLastProvenEpoch[setId]
-        );
+    function calculateProofFeeForSize(uint256 rawSize) public view returns (uint256) {
+        require(rawSize > 0, "failed to validate: raw size must be greater than 0");
+
+        // Use current fee or proposed fee if it's effective
+        uint256 currentFeePerTiB =
+            (block.timestamp >= feeEffectiveTime && proposedFeePerTiB > 0) ? proposedFeePerTiB : feePerTiB;
```

Comment:
check `feeEffectiveTime > 0` instead of `proposedFeePerTiB > 0` so that a proposed price of 0 would also trigger automatically